api/src/main/java/org/apache/commons/rdf/api/RDF.java

```diff
@@ -0,0 +1,244 @@
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.api;
+
+import java.io.Serializable;
+import java.util.Locale;
+
+/**
+ * A RDF implementation.
+ * <p>
+ * A <code>RDF</code> implementation can create instances of 
+ * the {@link RDFTerm} types
+ * {@link IRI}, {@link BlankNode} and {@link Literal}, as well as creating
+ * instances of the types {@link Triple}, {@link Quad}, {@link Graph} or
+ * {@link Dataset}.
+ * <p>
+ * A <em>partial RDF implementation</em> 
+ * should be clearly documented as such, and may throw 
+ * {@link UnsupportedOperationException} where applicable, e.g. if it 
+ * does not support creating {@link Dataset}s or {@link Quad}s. 
+ * <p>
+ * Instances of <code>RDF</code> work like a factory for 
+ * creating Commons RDF instances.
+ * spezializations of this interface may also provide methods  
+ * for conversions from/to their underlying RDF framework.
+ * <p>
+ * If a factory method of a particular implementation does not allow or support
+ * a provided parameter, e.g. because an IRI is considered invalid, then it
+ * SHOULD throw {@link IllegalArgumentException}.
+ *
+ * @since 0.3.0-incubating
+ * @see RDFTerm
+ * @see Graph
+ * @see Quad
+ */
+public interface RDF {
+
+    /**
+     * Create a new blank node.
+     * <p>
+     * The returned blank node MUST NOT be equal to any existing
+     * {@link BlankNode} instances according to {@link BlankNode#equals(Object)}.
+     *
+     * @return A new, unique {@link BlankNode}
+     */
+	 public BlankNode createBlankNode();
+
+    /**
+     * Create a blank node based on the given name.
+     * <p>
+     * All {@link BlankNode}s created with the given <code>name</code>
+     * <em>on a particular instance</em> of <code>RDF</code> MUST be
+     * equivalent according to {@link BlankNode#equals(Object)},
+     * <p>
+     * The returned BlankNode MUST NOT be equal to <code>BlankNode</code>
+     * instances returned for any other <code>name</code> or those returned from
+     * {@link #createBlankNode()}.
+     * <p>
+     * The returned BlankNode SHOULD NOT be equivalent to any BlankNodes created
+     * on a <em>different</em> <code>RDF</code> instance, e.g.
+     * different instances of <code>RDF</code> should produce
+     * different blank nodes for the same <code>name</code> unless they
+     * purposely are intending to create equivalent {@link BlankNode}
+     * instances (e.g. a reinstated {@link Serializable} factory).
+     *
+     * @param name
+     *            A non-empty, non-null, String that is unique to this blank
+     *            node in the context of this {@link RDF}.
+     * @return A BlankNode for the given name
+     */
+	public BlankNode createBlankNode(String name);
+
+    /**
+     * Create a new graph.
+     *
+     * It is undefined if the graph will be persisted by any underlying storage
+     * mechanism.
+     *
+     * @return A new Graph
+     */
+	 public Graph createGraph();
```

Comment:
Thanks, I'll do a separate commit to fix indentation to all spaces, which I think is what Commons prefer. (That got quite a large diff, so I won't pollute this PR with it)